File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/SQLQueryBuilder.cpp
#include <string>
#include <vector>
#include <utility>

class SQLQueryBuilder {
public:
    using Dict = std::vector<std::pair<std::string, std::string>>;

    static std::string select(const std::string& table, const std::vector<std::string>& columns = {"*"}, const Dict& where = {}) {
        std::string cols_str;
        for (size_t i = 0; i < columns.size(); ++i) {
            if (i > 0) cols_str += ", ";
            cols_str += columns[i];
        }
        std::string query = "SELECT " + cols_str + " FROM " + table;
        if (!where.empty()) {
            query += " WHERE ";
            for (size_t i = 0; i < where.size(); ++i) {
                if (i > 0) query += " AND ";
                query += where[i].first + "='" + where[i].second + "'";
            }
        }
        return query;
    }

    static std::string insert(const std::string& table, const Dict& data) {
        std::string keys, values;
        for (size_t i = 0; i < data.size(); ++i) {
            if (i > 0) {
                keys += ", ";
                values += ", ";
            }
            keys += data[i].first;
            values += "'" + data[i].second + "'";
        }
        return "INSERT INTO " + table + " (" + keys + ") VALUES (" + values + ")";
    }

    static std::string delete_(const std::string& table, const Dict& where = {}) {
        std::string query = "DELETE FROM " + table;
        if (!where.empty()) {
            query += " WHERE ";
            for (size_t i = 0; i < where.size(); ++i) {
                if (i > 0) query += " AND ";
                query += where[i].first + "='" + where[i].second + "'";
            }
        }
        return query;
    }

    static std::string update(const std::string& table, const Dict& data, const Dict& where = {}) {
        std::string update_str;
        for (size_t i = 0; i < data.size(); ++i) {
            if (i > 0) update_str += ", ";
            update_str += data[i].first + "='" + data[i].second + "'";
        }
        std::string query = "UPDATE " + table + " SET " + update_str;
        if (!where.empty()) {
            query += " WHERE ";
            for (size_t i = 0; i < where.size(); ++i) {
                if (i > 0) query += " AND ";
                query += where[i].first + "='" + where[i].second + "'";
            }
        }
        return query;
    }
};

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_py/code_output/SQLQueryBuilder.py
import re


class SQLQueryBuilder:

    @staticmethod
    def select(table, columns, where=None):
        if columns is None:
            columns = "*"
        if isinstance(columns, str):
            columns = re.split(r',\s*', columns)

        if columns:
            col_str = ", ".join(columns)
        else:
            col_str = "*"

        query = f"SELECT {col_str} FROM {table}"

        if where:
            where_parts = []
            for key, value in where.items():
                where_parts.append(f"{key}='{value}'")
            query += " WHERE " + " AND ".join(where_parts)

        return query

    @staticmethod
    def insert(table, data):
        keys = []
        values = []
        for key, value in data.items():
            keys.append(key)
            values.append(f"'{value}'")

        return f"INSERT INTO {table} ({', '.join(keys)}) VALUES ({', '.join(values)})"

    @staticmethod
    def delete(table, where=None):
        query = f"DELETE FROM {table}"

        if where:
            where_parts = []
            for key, value in where.items():
                where_parts.append(f"{key}='{value}'")
            query += " WHERE " + " AND ".join(where_parts)

        return query

    @staticmethod
    def update(table, data, where=None):
        set_parts = []
        for key, value in data.items():
            set_parts.append(f"{key}='{value}'")

        query = f"UPDATE {table} SET {', '.join(set_parts)}"

        if where:
            where_parts = []
            for key, value in where.items():
                where_parts.append(f"{key}='{value}'")
            query += " WHERE " + " AND ".join(where_parts)

        return query